ReactWindows/ReactNative.Shared/Views/View/ReactViewManager.cs

```diff
@@ -30,10 +31,57 @@ private enum Radius
             BottomRight,
         }
 
+        private class BorderProps
+        {
+            public uint? Color;
+        }
+
+        private readonly ConditionalWeakTable<BorderedCanvas, BorderProps> _borderProps =
+            new ConditionalWeakTable<BorderedCanvas, BorderProps>();
+
+        /// <summary>
+        /// Some border props do not affect appearance unless they are combined
+        /// with other props. Such props are saved in the dictionary and applied
+        /// later. This allows to not create the inner Border if it's not visible.
+        /// </summary>
+        private BorderProps GetBorderProps(BorderedCanvas view)
+        {
+            BorderProps props;
+
+            if (!_borderProps.TryGetValue(view, out props))
+            {
+                props = new BorderProps();
+                _borderProps.Add(view, props);
+            }
+
+            return props;
+        }
+
+        /// <summary>
+        /// Canvas.Background supports flat backgrounds. Border.Background supports
+        /// backgrounds with customizations, such as rounded corners. If the background
+        /// is flat, it's set on Canvas. If it has cutomizations, it's transferred to Border.
+        /// </summary>
+        private void TransferBackgroundBrush(BorderedCanvas view)
+        {
+            if (view.Background != null && view.Background != _defaultBackgroundBrush && view.Border != null)
+            {
+                view.Border.Background = view.Background;
+                view.Background = _defaultBackgroundBrush;
+            }
+        }
+
         /// <summary>
         /// Default brush for the view borders.
         /// </summary>
-        protected static readonly Brush s_defaultBorderBrush = new SolidColorBrush(Colors.Black);
+        protected readonly Brush _defaultBorderBrush = new SolidColorBrush(Colors.Black);
+
+        /// <summary>
+        /// In WPF in order to be clickable (hit-test visible) the element needs to have background brush.
+        /// This is why when the background and border brushes are set on the inner Border, the Canvas gets
+        /// a transparent background brush.
+        /// </summary>
+        protected readonly Brush _defaultBackgroundBrush = new SolidColorBrush(Color.FromArgb(0, 0, 0, 0));
 
```

Comment:
I'd rather not have the implementation details of workaround for WPF leak into UWP.

Reply:
We can either create two copies of this code (one that lives in the UWP solution and one in the WPF), or we can add precompiler directives. My preference would probably be the latter, since the goal in general is to move closer to a core portable library with cleaner abstractions and fewer instances of shared code.

---
In reply to: [154767340](https://github.com/Microsoft/react-native-windows/pull/1521#discussion_r154767340) [](ancestors = 154767340)

Reply:
I've added an `ifdef` for now. Are you suggesting to duplicate the entire file or introduce something like a solution-specific property that can be used in this file? e.g. the WPF solution could have a definition like `DefBgBrush = ...`, the UWP one would have `DefBgBrush = null` and `ReactViewManager` would use the `DefBgBrush` variable.